javaee/impl/src/main/java/org/jboss/forge/addon/javaee/security/ui/AddRoleCommand.java

```diff
@@ -0,0 +1,65 @@
+package org.jboss.forge.addon.javaee.security.ui;
+
+import org.jboss.forge.addon.facets.constraints.FacetConstraint;
+import org.jboss.forge.addon.javaee.servlet.ServletFacet;
+import org.jboss.forge.addon.javaee.ui.AbstractJavaEECommand;
+import org.jboss.forge.addon.ui.command.PrerequisiteCommandsProvider;
+import org.jboss.forge.addon.ui.context.UIBuilder;
+import org.jboss.forge.addon.ui.context.UIContext;
+import org.jboss.forge.addon.ui.context.UIExecutionContext;
+import org.jboss.forge.addon.ui.input.UIInput;
+import org.jboss.forge.addon.ui.metadata.UICommandMetadata;
+import org.jboss.forge.addon.ui.metadata.WithAttributes;
+import org.jboss.forge.addon.ui.result.NavigationResult;
+import org.jboss.forge.addon.ui.result.Result;
+import org.jboss.forge.addon.ui.result.Results;
+import org.jboss.forge.addon.ui.result.navigation.NavigationResultBuilder;
+import org.jboss.forge.addon.ui.util.Categories;
+import org.jboss.forge.addon.ui.util.Metadata;
+
+import javax.inject.Inject;
+
+/**
+ * Adds a security role to the current project.
+ *
+ * @author <a href="mailto:ivan.st.ivanov@gmail.com">Ivan St. Ivanov</a>
+ */
+@FacetConstraint(ServletFacet.class)
+public class AddRoleCommand extends AbstractJavaEECommand
```

Comment:
The remove role command along with its test is ready. But I am not completely sure about the remove security constraint. The thing is that the API is not very convenient for removing or even manipulating single elements. One can remove all the constraints, get all the constraints and add a constraint (or get the first one). 

So what I could think of is create a copy of all the existing constraints, then delete them from the web.xml, then add them one by one by skipping the one that we want to remove. It is not very elegant (especially with pre Java 8 code). This was fine with security roles, which is relatively simple type to re-create. However, the security constraints are quite complex: they contain web resource collection, authentication constraints and user data constraints. Which are complex types with the same kind of API challenges.

To conclude my tirade, I propose to not implement remove constraint command to avoid writing ugly and hard to maintain code. Looking at the other addons, there are not many remove commands anyway. Forge excels at tasks where one wants to create something, because they don't know its syntax or the exact rules. Once you have it, removing is child's play :)